sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubMessageSerializer.java

```diff
@@ -107,14 +118,85 @@ public int getSize(Message amqpMessage) {
 
         final EventData eventData = (EventData) object;
         final Message message = Proton.message();
+        final byte[] body = eventData.getBody();
+
+        //TODO (conniey): support AMQP sequence and AMQP value.
+        message.setBody(new Data(new Binary(body)));
 
         if (eventData.getProperties() != null && !eventData.getProperties().isEmpty()) {
             message.setApplicationProperties(new ApplicationProperties(eventData.getProperties()));
         }
 
-        setSystemProperties(eventData, message);
+        if (message.getProperties() == null) {
+            message.setProperties(new Properties());
+        }
+
+        message.setMessageId(eventData.getMessageId());
+        message.setContentType(eventData.getContentType());
+        message.setCorrelationId(eventData.getCorrelationId());
+        message.setSubject(eventData.getRawAmqpMessage().getProperties().getSubject());
+        message.setReplyTo(eventData.getRawAmqpMessage().getProperties().getReplyTo().toString());
+        message.setReplyToGroupId(eventData.getRawAmqpMessage().getProperties().getReplyToGroupId());
+        message.setGroupId(eventData.getRawAmqpMessage().getProperties().getGroupId());
+
+        final AmqpMessageProperties brokeredProperties = eventData.getRawAmqpMessage().getProperties();
+
+        message.setContentEncoding(brokeredProperties.getContentEncoding());
+        if (brokeredProperties.getGroupSequence() != null) {
+            message.setGroupSequence(brokeredProperties.getGroupSequence());
+        }
+        message.getProperties().setTo(eventData.getRawAmqpMessage().getProperties().getTo().toString());
```

Comment:
nit:  Same observation/question about calling into `message.getProperties()` over and over.

Reply:
According to your instructions, fix in new version.